TypeScript (#1819)
enkidevs/curriculum: typescript/introduction/tsc-variable-declaration/tsc-const.md

---
author: kapnobatai136

type: normal

category: must-know

aspects:
  - introduction

---

# Introducing `const`

---
## Content

The only difference between `const` and `let` is that the variables declared with `const` cannot be reassigned:

> 

Reviewer: ```suggestion

> *Note*: Assignment is done using the `=` operator. 

```

```ts
// because its declared with

> // let, reassigning a to a

Reviewer: ```suggestion
// let, reassigning to a
```

// new value is fine
let a: number = 1;
a = 2;

// on the other hand, b is
// declared with const and
// cannot be reassigned
const b: string = 'enki';
b = 'foobar'; // Error
```

Although variables declared with `const` can't be reassigned, that doesn't mean that the values they hold are constants and can't be changed.

```ts
const c: number[] = [
  1,
  2,
  3
];
c.push(4);
console.log(c);
// [1, 2, 3, 4]
```

> When using `const`, the variable name is constant once declared but not the value.

Reviewer: ```suggestion
When using `const`, the assignment can happen only once but the value can be changed whenever.
```
